### ISSUE TITLE
lastmod not set for pages rendered as a folder

### ISSUE THREAD

**Issue by jayvdb** [jekyll/jekyll-sitemap#208]
If the page permalink does *not* contain an extension, such as `.html` / `:output_ext` , jekyll creates a folder with an `index.html` inside, and `jekyll-sitemap` only emits a `loc` without a `lastmod`.

**Comment by jayvdb** [comment]
Actually it seems like lots of 'pages' (i.e. files with frontmatter) are not given a lastmod, and we need to use `jekyll-last-modified-at`.

That seems very unnecessary, and a default fallback of using the file mtime would be sufficient for most people.

**Comment by pathawks** [comment]
> That seems very unnecessary, and a default fallback of using the file mtime would be sufficient for most people.

The big problem here is that GitHub Pages only gets a shallow clone of the repository, so _everything_ would have the same mtime. Any work-around is beyond the scope of this plugin, which is why we rely on **jekyll-last-modified-at**.

As far as not checking `index.html` when the permalink is a directory, that sounds like a real issue.

**Comment by jayvdb** [comment]
> As far as not checking index.html when the permalink is a directory, that sounds like a real issue.

When I add `jekyll-last-modified-at`, the modified time is set to the appropriate value.

> The big problem here is that GitHub Pages only gets a shallow clone of the repository, so everything would have the same mtime.

Yea, that is a problem with any git clone, not just shallow clones. (I checked GitLab Pages has the same).  That means my "would be sufficient for most people" is likely not correct.

That said, it is core jekyll functionality to preserve mtimes.  See https://github.com/jekyll/jekyll/pull/3220

This will help anyone who is doing builds on Travis CI and pushing the results to a `gh-pages` branch (or similar).  But that approach allows custom gems, so doesnt need to be solved in this core GitHub Pages gem.

So, I've raised an issue over at https://github.com/gjtorikian/jekyll-last-modified-at/issues/54 .
